### PR TITLE
Allow deletion of subscriptions based on filters.

### PR DESCRIPTION
**Pull request by mmitche**
Deprecate delete-subscription and replace with delete-subscriptions (same parameter set supported, plus all the filters of get-subscriptions and trigger subscriptions). This will allow easier cleanup of large numbers of subscriptions.

### CODE REVIEW

File: src/Maestro/tests/Scenarios/common.ps1 [hunk 1]
```diff
@@ -166,7 +166,7 @@ function Darc-Command-WithPipeline([Parameter(ValueFromPipeline=$true)]$pipeline
     if ($darcCommandPrefix) {
         $finalParams = ,"$darcCommandPrefix" + $finalParams
     }
-    Write-Host "Running 'pipelineParams | $darcTool $finalParams $darcAuthParams'"
+    Write-Host "Running 'pipelineParams | $darcTool $finalParams ***'"
```
Comment (riarenas) [inline src/Maestro/tests/Scenarios/common.ps1]: woops

Comment (mmitche) [reply src/Maestro/tests/Scenarios/common.ps1]: I wasn't terribly worried about it, just that it was very noisy in the output

File: src/Microsoft.DotNet.Darc/src/Darc/Options/DeleteSubscriptionsCommandLineOptions.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+using CommandLine;
+using Microsoft.DotNet.Darc.Operations;
+
+namespace Microsoft.DotNet.Darc.Options
+{
+    [Verb("delete-subscriptions", HelpText = "Delete a subscription or set of subscriptions matching criteria.")]
+    internal class DeleteSubscriptionsCommandLineOptions : SubscriptionsCommandLineOptions
+    {
+        [Option("id", HelpText = "Delete a specific subscription by id.")]
+        public string Id { get; set; }
```
Comment (riarenas) [inline src/Microsoft.DotNet.Darc/src/Darc/Options/DeleteSubscriptionsCommandLineOptions.cs]: Should we also make this Ids like in get-subscriptions?

Comment (mmitche) [reply src/Microsoft.DotNet.Darc/src/Darc/Options/DeleteSubscriptionsCommandLineOptions.cs]: Id's is supported in the base class

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/DeleteSubscriptionsOperation.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                    }
+                    await remote.DeleteSubscriptionAsync(subscription.Id.ToString());
+                }
+                Console.WriteLine($"done");
+
+                return Constants.SuccessCode;
+            }
+            catch (Exception e)
+            {
+                Logger.LogError(e, "Unexpected error while delete subscriptions.");
```
Comment (riarenas) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/DeleteSubscriptionsOperation.cs]:
```suggestion
                Logger.LogError(e, "Unexpected error while deleting subscriptions.");
```

**Pull request merged**